table/sst_file_writer.cc

```diff
@@ -181,15 +198,31 @@ Status SstFileWriter::Finish(ExternalSstFileInfo* file_info) {
     r->ioptions.env->DeleteFile(r->file_info.file_path);
   }
 
-  if (s.ok() && file_info != nullptr) {
-    r->file_info.file_size = r->builder->FileSize();
+  if (file_info != nullptr) {
     *file_info = r->file_info;
   }
 
   r->builder.reset();
   return s;
 }
 
+void SstFileWriter::InvalidatePageCache(bool closing) {
+  Rep* r = rep_;
+  if (r->fadvise_trigger == 0) {
+    // Fadvise disabled
+    return;
+  }
+
+  uint64_t bytes_in_page_cache = r->builder->FileSize() - r->last_fadvise_size;
+  if (bytes_in_page_cache > r->fadvise_trigger ||
+      (closing && bytes_in_page_cache > 0)) {
+    TEST_SYNC_POINT_CALLBACK("SstFileWriter::InvalidatePageCache",
+                             &(bytes_in_page_cache));
+    r->file_writer->InvalidateCache(r->last_fadvise_size, bytes_in_page_cache);
```

Comment:
@yoshinorim, I think the option that could affect us is Options::disableDataSync
but maybe @siying  is referring to another option since he is saying `sync per block`